salt/cloud/clouds/gce.py

```diff
@@ -387,12 +387,33 @@ def __get_size(conn, vm_):
     """
     Need to override libcloud to find the machine type in the proper zone.
     """
+    default_size = "n1-standard-1"
     size = config.get_cloud_config_value(
-        "size", vm_, __opts__, default="n1-standard-1", search_global=False
+        "size", vm_, __opts__, default=default_size, search_global=False
     )
+    if not size or not isinstance(size, str):
```

Comment:
Can we remove this as discussed previously. `get_cloud_config_value` will return the default if its not set.